File: SoccerTeamManagement/ClientApp/src/app/interfaces/person.ts
import { IAddress } from "./address";
import { IImage } from "./image";
import { IPhone } from "./phone";
import { ICountry } from "./lookups/country";
import { IEntity } from "./entity";

export interface IPerson extends IEntity {
  firstName: string;
  lastName: string;
  nickName: string;
  dateOfBirth: Date;

  phoneId: number;
  phone: IPhone;
  addressId: number;
  address: IAddress;
  imageId: number;
  image: IImage;
  countryId: number;
  country: ICountry;
}

File: SoccerTeamManagement/Data/Models/People/Person.cs
using System.Collections.Generic;

namespace SoccerTeamManagement.Data.Models.People
{
    public abstract class Person : EntityBase
    {
        #region Properties

        public string FirstName { get; set; }

        public string LastName { get; set; }

        public string NickName { get; set; }

        public bool IsAdult { get; set; }

        #endregion Properties

        #region Relationships

        public int? PhoneId { get; set; }

        public Phone Phone { get; set; }

        public int? AddressId { get; set; }

        public Address Address { get; set; }

        public int? ImageId { get; set; }

        public Image Image { get; set; }

        public int? NationId { get; set; }

        public CountryLookup Nation { get; set; }

        #endregion Relationships
    }
}